src/sinks/util/tcp.rs

```diff
@@ -77,20 +78,28 @@ pub struct TcpSink {
     resolver: Resolver,
     tls: MaybeTlsSettings,
     state: TcpSinkState,
+    // state2: TcpSinkState2,
     backoff: ExponentialBackoff,
     span: tracing::Span,
 }
 
 enum TcpSinkState {
     Disconnected,
     ResolvingDns(crate::dns::ResolverFuture),
-    Connecting(MaybeTlsConnector),
+    Connecting(Box<dyn Future<Item = MaybeTlsStream<TcpStream>, Error = TlsError> + Send>),
     Connected(TcpOrTlsStream),
     Backoff(Box<dyn Future<Item = (), Error = ()> + Send>),
 }
 
-type TcpOrTlsStream =
-    CompatSink<FramedWrite<Compat<Compat01As03<MaybeTlsStream<TcpStream>>>, BytesCodec>, Bytes>;
+type TcpOrTlsStream = CompatSink<FramedWrite<MaybeTlsStream<TcpStream>, BytesCodec>, Bytes>;
+
+// enum TcpSinkState2 {
+//     Disconnected,
+//     ResolvingDns(BoxFuture<'static, Result<crate::dns::LookupIp, crate::dns::DnsError>>),
+//     Connecting(BoxFuture<'static, Result<MaybeTlsStream<TcpStream>, TlsError>>),
+//     Connected(FramedWrite<MaybeTlsStream<TcpStream>, BytesCodec>),
+//     Backoff(BoxFuture<'static, ()>),
+// }
```

Comment:
We should clean up all the commented out code before merging.

Reply:
Yeah, I understand that this code not related anyhow with current TcpSink, but can be very useful in future for saving time of anybody who will work on migrating to new Sink trait. If you insist to delete this, can I at least add a link to relevant code in this PR with a comment about new Sink trait impl?